coderd/templateversions.go

```diff
@@ -1742,3 +1812,29 @@ func (api *API) publishTemplateUpdate(ctx context.Context, templateID uuid.UUID)
 			slog.F("template_id", templateID), slog.Error(err))
 	}
 }
+
+func checkProvisioners(ctx context.Context, store database.Store, orgID uuid.UUID, wantTags map[string]string, pollInterval time.Duration) (found, active int, err error) {
+	// Check for eligible provisioners. This allows us to return a warning to the user if they
+	// submit a job for which no provisioner is available.
+	eligibleProvisioners, err := store.GetProvisionerDaemonsByOrganization(ctx, database.GetProvisionerDaemonsByOrganizationParams{
+		OrganizationID: orgID,
+		WantTags:       wantTags,
+	})
+	if err != nil {
+		// Log the error but do not return any warnings. This is purely advisory and we should not block.
+		return -1, -1, xerrors.Errorf("get provisioner daemons by organization: %w", err)
+	}
+
+	onePollAgo := time.Now().Add(-pollInterval)
```

Comment:
How much is the poll interval/hearbeat? If it's 1:1 this may be a bit too aggressive. Perhaps we should remove an additional 30 seconds - 1 minute?

Reply:
We actually stop showing provisioners more than 3 poll intervals old in the UI if memory serves, so it might be good to align on that.